### PR TITLE
Fixes #1001: Added error handling and retry service

### LINKED ISSUE #1001: Use Error handling for some services and make request again in case of an error.
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behavior:**
Error handling is not present for some of the services like `autocorrect` and `autocomplete`.
**Expected behavior:**

Implement error handling and `retry` for the services in case of an error.

### PR DESCRIPTION
**Pull request by praveenojha33**
Fixes #1001 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:
Implement error handling and retry for the services in case of an error.
Surge Link:https://pr-1002-fossasia-susper.surge.sh

### CODE REVIEW

**Comment by praveenojha33** [comment]
@simasausaurabh Please review.

File: src/app/services/search.service.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ export class SearchService {
 
         res.json()[0]
 
-      ).catch(this.handleError);
+      )
+      .catch(this.handleError);
```
Comment (simsausaurabh) [inline src/app/services/search.service.ts]: No change here.

File: src/app/services/speech-synthesis.service.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { Injectable, NgZone } from '@angular/core';
 import { Observable } from 'rxjs/Rx';
 
+
```
Comment (simsausaurabh) [inline src/app/services/speech-synthesis.service.ts]: Unwanted change.

**Comment by praveenojha33** [comment]
@simsausaurabh I have done the required changes please review now.

**Pull request merged**